pkg/codegen/schema/schema.go

```diff
@@ -743,7 +743,8 @@ func (pkg *Package) MarshalJSON() (bytes []byte, err error) {
 				continue
 			}
 
-			o, err := pkg.marshalObject(t)
+			// Use the input shape when marshaling in order to get the plain annotations right.
+			o, err := pkg.marshalObject(t.InputShape, false)
```

Comment:
These booleany call sites will be tough reading some day. `/* plain */ false` or `marshalObjectInputty`?